### ISSUE TITLE
Feature request: Disallow parent selectors

### ISSUE THREAD

**Issue by eldh** [brigade/scss-lint#758]
I'd like to see a rule to disallow some parent selectors, like this:

Bad: 
```
.Base {
  background-color: red;
  &__element {
    background-color: blue;
  }
}
```

Good:
```
.Base {
  background-color: red;
  &:after {
    background-color: blue;
  }
}
```

Good: 
```
.Base {
  background-color: red;
}
.Base__element {
  background-color: blue;
}
```

Is this somehow already possible? Does it make sense as a rule? Is it possible?

**Comment by abea** [comment]
This seems to conflict with the `::after` pseudo element, which can still be written with one colon. I'd expect your second sample to compile to:
```css
.Base {
  background-color: red;
}
.Base:after {
  background-color: blue;
}
```

Regardless, I don't think disallowing parent selectors is possible right now. Since there isn't an alternative to `&__element`, I don't think I could see using this in production.

**Comment by eldh** [comment]
@abea The alternative would be using the syntax from my third example. 

The reason for this is that I find the `&__element` syntax can get really messy for bigger modules, and explicitly defining class names makes sure you can always search for a class name to see it's rules.

**Comment by abea** [comment]
Ah, I misunderstood the examples. I can definitely see that.

**Comment by eldh** [comment]
@abea Sorry that I was unclear!

**Comment by svyatov** [comment]
Yeah, that's a good idea. I liked this syntax at first, but it absolutely ruins search possibilities.

@eldh although, I think your second example is completely irrelevant and confusing here :)

**Comment by eldh** [comment]
Yeah, I just wanted to clarify that the rule shouldn't apply to all uses of `&`, but I see how it can be confusing.

**Comment by eldh** [comment]
I edited the description now to hopefully make it more clear .

**Comment by shukriadams** [comment]
I had the same need, so I hacked one together at https://github.com/shukriadams/selectorConcatenation. It was done in a hurry but it seems to work. Basically, any selector that starts with & followed by letters, numbers dashes or underscores gets flagged. Too simplistic?

And totally agree, this is one Sass feature that has created huge headaches for us when it comes to code maintenance.
